text/0258-env-context-baggage-carriers.md

```diff
@@ -0,0 +1,236 @@
+Environmental Variables can be used.
+
+Consider the following diagram:
+
+![Environment Variable Context Propagation](./img/0258-env-context-parent-child-process.png)
+
+In this diagram, a parent process is forked to spawn a child process,
+inheriting the environment variables from the original parent. The environment
+variables defined here, `TRACEPARENT`, `TRACESTATE`, and `BAGGAGE` are used to
+propagate context to the child process such that it can be tied to the parent.
+Without `TRACEPARENT`, a tracing backend would not be able to connect the child
+process spans to the parent span, forming an end-to-end trace.
+
+`traceparent` (lowercase), originates in the [W3C Specification][w3c-parent]
+and includes the following fields:
+
+* `version`
+* `trace-id`
+* `parent-id`
+* `trace-flags`
+
+`tracestate` (lowercase), originates in the [W3C Specification][w2c-state] and
+can include any opaque value in a key-value pair structure. Its goal is to
+provide additional vendor-specific trace information.
+
+`baggage` (lowercase), also is defined in the [W3C Specification][w3c-bag] and
+is a set of key-value pairs to propagate context between signals. In
+OpenTelemetry, baggage is propagated through the [Baggage API][bag-api].
+
+Consider this real world example OpenTofu Controller Deployment.
+
+![OpenTofu Run](./img/0258-env-context-opentofu-tracing.png)
+
+In this model, the OpenTofu Controller is the start of the trace, containing
+the actual trace_id and generating the root span. The OpenTofu controller
+deploys a runner which has it's own environment and processes to run OpenTofu
+commands. If one was to trace these processes without a carrier mechanism, then
+they would all show up as unrelated root spans as separate traces. However, by
+leveraging environment variables as carriers, each span is able to be tied back
+to the root span, creating a single trace as shown in the image of a real
+OpenTofu trace below.
+
+![OpenTofu Trace](./img/0258-env-context-opentofu-trace.png)
+
+Additionally, the `init` span is able to pass baggage to the `plan` and `apply`
+spans. One example of this is module version and repository information. This
+information is only determined and known during the `init` process. Subsequent
+processes only know about the module by name. With `BAGGAGE` the rest of the
+proccesses are able to understand a key piece of information which allows
+errors to be tied back to original module version and source code.
+
+Defining the specification for Environment Variables as carriers will have a
+wide impact to the industry in enabling better observability to systems outside
+of the normal HTTP microservice architecture.
+
+[w3c-parent]: https://www.w3.org/TR/trace-context/#traceparent-header
+[w3c-state]: https://www.w3.org/TR/trace-context/#tracestate-header
+[w3c-bag]: https://www.w3.org/TR/baggage/#header-name
+[bag-api]: https://opentelemetry.io/docs/specs/otel/baggage/api/
+
+## Internal details
```

Comment:
I think this should be the core of the proposed spec changes, not "internal details". The spec should be for env-var carrier, which should describe how key-value pairs produced by an arbitrary TextMapPropagator can be encoded in a safe and lossless way in env variables.

Reply:
+1

Reply:
I updated the core of the spec to encompass this. Let me know if this is the direction you're thinking. Im going to be re-reading and making further edits to, but wanted to gauge direction.